File: NoammAddons/features/AbilityKeybinds.js
/// <reference types="../../CTAutocomplete" />
/// <reference lib="es2015" />

import Settings from "../Settings"
import PogObject from "../../PogData"
import { PlayerUtils, IsInDungeon } from "../utils"


const Keybinds = new PogObject("Noammaddons", {
	ClassULTIMATE: Keyboard.KEY_NONE,
	ClassAbility: Keyboard.KEY_NONE
}, "Config/Keybinds.json")


let ClassULTIMATE = new KeyBind("Class ULTIMATE", Keybinds.ClassULTIMATE, "NoammAddons")
let ClassAbility = new KeyBind("Class ABILITY", Keybinds.ClassAbility, "NoammAddons")




ClassULTIMATE.registerKeyPress(() => {
    if (!Settings.AbilityKeybinds || !IsInDungeon()) return
    PlayerUtils.UseDungeonClassAbility(true)
    Keybinds.ClassULTIMATE = ClassULTIMATE.getKeyCode()
})
    
ClassAbility.registerKeyPress(() => {
    if (!Settings.AbilityKeybinds || !IsInDungeon()) return
    PlayerUtils.UseDungeonClassAbility(false)
    Keybinds.ClassAbility = ClassAbility.getKeyCode()
})
    

File: src/main/kotlin/noammaddons/features/impl/dungeons/AbilityKeybinds.kt
package noammaddons.features.impl.dungeons

import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
import noammaddons.events.PreKeyInputEvent
import noammaddons.features.Feature
import noammaddons.ui.config.core.impl.*
import noammaddons.utils.DungeonUtils.dungeonStarted
import noammaddons.utils.LocationUtils.inDungeon
import noammaddons.utils.PlayerUtils.useDungeonClassAbility

object AbilityKeybinds: Feature("Allows you do use your dungeon class ult/ability with a keybind") {
    val classUltimate = ToggleSetting("Class Ultimate", true)
    val classAbility = ToggleSetting("Class Ability", true)
    val ultKeybind = KeybindSetting("Ultimate Keybind")
    val abilityKeybind = KeybindSetting("Ability Keybind")

    override fun init() {
        ultKeybind.addDependency(classUltimate)
        abilityKeybind.addDependency(classAbility)

        addSettings(
            classUltimate, classAbility,
            SeperatorSetting("Keybinds").addDependency { ! classUltimate.value || ! classAbility.value },
            ultKeybind, abilityKeybind
        )
    }

    @SubscribeEvent
    fun onKeyInput(event: PreKeyInputEvent) {
        if (! inDungeon || ! dungeonStarted) return
        if (classUltimate.value && ultKeybind.isPressed()) useDungeonClassAbility(true)
        if (classAbility.value && abilityKeybind.isPressed()) useDungeonClassAbility(false)
    }
}
